packages/diffhtml/lib/tree/sync.js

```diff
@@ -71,7 +71,7 @@ export default function syncTree(oldTree, newTree, patches, parentTree, specialC
     // then splice it into the parent (if it exists) and run a sync.
     if (retVal && retVal !== newTree) {
       newTree.childNodes = [].concat(retVal);
-      syncTree(null, retVal, patches, newTree);
+      syncTree(oldTree !== empty ? oldTree : null, retVal, patches, newTree);
```

Comment:
Okay maybe we should add a comment here that we're using the `oldTree` if it exits to short-circuit renders. The reason for my change was my thinking that this would only ever be used to find attribute changes, hence the `null`. I should have added comment myself. I'm :+1: to this change. Thanks!